### ISSUE TITLE
Escaping from a custom menu loses focus

### ISSUE THREAD

**Issue by chrisjj** [microsoft/vscode#90729]
1 Tab to this
![image](https://user-images.githubusercontent.com/6157073/74575915-bcca3c80-4f80-11ea-9b2c-8a98ffc36d7a.png)
2 Press Enter
![image](https://user-images.githubusercontent.com/6157073/74575956-e6836380-4f80-11ea-9280-2732c8519e7c.png)
3 Press Esc

Expected: focus returns to '...'
Observed: '...' button disappears and the focus moves to somewhere invisible.
![image](https://user-images.githubusercontent.com/6157073/74576055-47ab3700-4f81-11ea-8d5c-1d9306282d4a.png)

V 1.42.1 on Windows 7.

**Comment by isidorn** [comment]
This works fine on the mac with native context menus.
The problem is windows and custom menus. Thus renaming and forwarding to @sbatten

**Comment by chrisjj** [comment]
Thanks. Any idea why Windows native menus are not used?

**Comment by isidorn** [comment]
They have a lot of issues, for example with accessiblity, thus we use custom menus on Windows.

**Comment by chrisjj** [comment]
Thanks. I'm surprised Mac has fewer accessibility issues.

**Comment by sbatten** [comment]
same as #90730, cannot reproduce in master

**Comment by eamodio** [comment]
I can reproduce this on master

**Comment by sbatten** [comment]
got it, always show actions setting breaks repro.

**Comment by isidorn** [comment]
Somewhat similar https://github.com/microsoft/vscode/issues/91085
When you press esc the focus should ideally return to the outer continer (like the tree does). If that is not possible the focus should not move and stay put.

**Comment by isidorn** [comment]
Actually that might be the same issue since the custom menu extends from action bar.
What I suggest is the following: in [`IActionBarOptions`](https://github.com/microsoft/vscode/blob/614a35cd23620fca7b62fc87ea93d91b1efd5be4/src/vs/base/browser/ui/actionbar/actionbar.ts#L398) we introduce an optional attribute `focusOnBlur: HTMLElement` that the creators of the `actionBar` can pass in.
The ActionBar would then pass focus to that element when user presses `esc`.
Alternative is that all `ActionBar` users need to listen to `onDidCancel` event and when the event fires they would pass focus to the appropriate element - similar to what the `menu` already does.
There are a lot of `ActionBar` clients and I think all of them need this behavior fixed.

Alternative is just to make esc a no-op in non menu use cases. However I think passing focus to the outer element is actually what the users might expect as @jvesouza points out

Let me know what you think @sbatten @bpasero

**Comment by chrisjj** [comment]
> we introduce an optional attribute `focusOnBlur: HTMLElement` that the creators of the `actionBar` can pass in.

Can't you stack the previous holder and then restore?

> Alternative is just to make esc a no-op in non menu use cases. 

I vote against.

**Comment by bpasero** [comment]
> Alternative is just to make esc a no-op in non menu use cases.

> I vote against.

Also, -1 for that. It is relatively simple: check what the OS does when you open a menu and press ESC and mimic that. After all we try to make the menu as close to the native experience as possible.

**Comment by sbatten** [comment]
fix is likely going to have to involve this https://github.com/microsoft/vscode/issues/93415#issuecomment-620091022

**Comment by chrisjj** [comment]
Closed why?

**Comment by sbatten** [comment]
@chrisjj The issue has been resolved with https://github.com/microsoft/vscode/commit/01b37c6b5d991f959f0b31a58055fecf6bc52729

**Comment by chrisjj** [comment]
>
> @chrisjj <https://github.com/chrisjj> The issue has been resolved with
> 01b37c6
> <https://github.com/microsoft/vscode/commit/01b37c6b5d991f959f0b31a58055fecf6bc52729>
>

Great. Thanks.
